src/Tribe/API.php

```diff
@@ -271,10 +271,14 @@ protected static function prepare_event_date_meta( $event_id, $data ) {
 
 			if ( isset( $data['EventStartDate'] ) ) {
 				$data['EventStartDate'] = Tribe__Date_Utils::datetime_from_format( $datepicker_format, $data['EventStartDate'] );
+			} elseif ( $existing_start_date = get_post_meta( $event_id, '_EventStartDate', true ) ) {
```

Comment:
I think it adds an overhead we do not need in this case (post id helper and such); it's good in "combat zone": by the time we are here we know the `$event_id` is legit